crates/re_build_tools/src/hashing.rs

```diff
@@ -0,0 +1,159 @@
+use std::fmt::Write;
+use std::path::{Path, PathBuf};
+use std::{fs, io};
+
+use anyhow::Context as _;
+use sha2::{Digest, Sha256};
+
+// ---
+
+fn encode_hex(bytes: &[u8]) -> String {
+    let mut s = String::with_capacity(bytes.len() * 2);
+    for &b in bytes {
+        write!(&mut s, "{b:02x}").unwrap();
+    }
+    s
+}
+
+/// Walks the directory at `path` in filename order.
+///
+/// If `extensions` is specified, only files with the right extensions will be iterated.
+/// Specified extensions should include the dot, e.g. `.fbs`.
+pub fn iter_dir<'a>(
+    path: impl AsRef<Path>,
+    extensions: Option<&'a [&'a str]>,
+) -> impl Iterator<Item = PathBuf> + 'a {
+    fn filter(entry: &walkdir::DirEntry, extensions: Option<&[&str]>) -> bool {
+        let is_dir = entry.file_type().is_dir();
+        let is_interesting = extensions.map_or(true, |extensions| {
+            extensions.iter().any(|ext| {
+                entry
+                    .file_name()
+                    .to_str()
+                    .map_or(false, |s| s.ends_with(ext))
```

Comment:
why not use `.extension` and compare that instead? Then the method doesn't need the "plz include the dot" constraint

Reply:
I don't mind either way. But now I'll have to specify "without the leading dot!" like std does ^_^